### ISSUE TITLE
[Bug] The email field in the Checkout block has `type:text` instead of `type: email`

### ISSUE THREAD

**Issue by ralucaStan** [woothemes/woocommerce#48610]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

<img width="565" alt="Screenshot 2024-06-19 at 12 44 06" src="https://github.com/woocommerce/woocommerce/assets/1628454/9b135d10-5522-4029-8022-8ea34f553533">
The email field in the Checkout block has `type:text`. 

This makes it harder for non-desktop users to fill in the information, as phone and tables optimize the keyboard depending on the input type. 

### Expected behavior

The fied should have `type:email` because:
- it's semantic 
- built in validation
- accesibility, better experience for screen readers
- browser features

### Actual behavior

The input is treated as a normal text field.

### Steps to reproduce

1. Go to the Checkout block
2. Inspect the Email address field under the Contact information section
3. Notice the type text
4. Go to a phone and notice that the email keyboard is not activated

### WordPress Environment

n/a

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.
